packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialog.tsx

```diff
@@ -0,0 +1,75 @@
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogHeader,
+  ScrollArea,
+  Text,
+  Video,
+} from "@prismicio/editor-ui";
+
+import {
+  useOnboardingProgress,
+  useOnboardingStepsContent,
+} from "@/features/onboarding/helpers";
+import type { OnboardingStep } from "@/features/onboarding/types";
+
+import styles from "./OnboardingStepDialog.module.css";
+
+type OnboardingStepDialogProps = {
+  step: OnboardingStep;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const OnboardingStepDialog = ({
+  step,
+  isOpen,
+  onClose,
+}: OnboardingStepDialogProps) => {
+  const { toggleStepComplete, isStepComplete, getStepIndex } =
+    useOnboardingProgress();
+  const content = useOnboardingStepsContent();
+  const stepContent = content[step.id];
+
+  const markAsDone = () => {
+    if (!isOpen) return;
+    toggleStepComplete(step.id);
+    onClose();
+  };
+
+  return (
+    <Dialog open={isOpen} onOpenChange={onClose}>
+      <DialogHeader title="Learn" />
+      <DialogContent>
+        <ScrollArea>
+          <article className={styles.scrollableContent}>
+            <section>
+              <Text
+                className={styles.stepNumberText}
```

Comment:
💡 #idea

I don't think you need this classname, see my message above about prevent using CSS modules in `features` folder.

I think you could just use `SX` prop from `Text` right?

Reply:
I don't particularly enjoy applying inline styles 😅 It's fine for one or two properties, but doesn't seem like a approach. Can you clarify why we shouldn't create CSS modules in `features`? 

There will always be the need for specific components, which don't make sense to have under a `components` directory because they are not used anywhere else. So for those ones, we will often need to apply some styles.

Reply:
That's the question, do you need CSS modules at all ever in features? One goal is to prevent specific components with specific style. If it's only Layout, then you `Box`. 
It's something PB introduced and prevent usage of styling directly in their wroom app. We could discuss this topic with them, I think they would give us better perspective on this.

Reply:
Refactored 🧼

Reply:
I could explain a bit more the vision next time we talk, and you can tell me what you think.